### ISSUE TITLE
SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.

### ISSUE THREAD

**Issue by adrianreber** [fedora-infra/mirrormanager2#84]
Seen on mm-frontend01 error_log:

``````
[:error] [pid 22412] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
[:error] [pid 22412]   cached_connections, mapper, table, delete)```
``````

**Comment by pypingou** [comment]
Not sure if this is a big issue, it failed to delete nothing :]
But agreed, would be nice to kick out anyway

**Comment by pypingou** [comment]
I was wondering if we ever managed to replicate this issue locally? (and if so, we should document the steps to reproduce it here)

**Comment by adrianreber** [comment]
I think this has only been seen in production. I grepped through the logfiles and it still exists on mm-frontend01:

```
[Mon Sep 28 10:12:22.333651 2015] [:error] [pid 32278] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
[Mon Sep 28 10:12:23.171836 2015] [:error] [pid 32399] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
[Mon Sep 28 17:07:47.169075 2015] [:error] [pid 2111] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
[Tue Sep 29 06:15:43.514388 2015] [:error] [pid 10831] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
[Tue Sep 29 06:15:44.304136 2015] [:error] [pid 10804] /usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py:117: SAWarning: DELETE statement on table 'host_category_dir' expected to delete 1 row(s); 0 were matched.  Please set confirm_deleted_rows=False within the mapper configuration to prevent this warning.
```

This is either from report_mirror or the web interface.

**Comment by pypingou** [comment]
Looking at git grep, I see host_category_dir being mostly used in the crawler

**Comment by adrianreber** [comment]
Should be fixed by #232
